data/basics/userBalance.go

```diff
@@ -167,16 +167,20 @@ type AppParams struct {
 
 // Clone returns a copy of some AppParams that may be modified without
 // affecting the original
-func (ap AppParams) Clone() (res AppParams) {
-	res = ap
+func (ap *AppParams) Clone() (res AppParams) {
+	res = *ap
+	res.ApprovalProgram = make([]byte, len(ap.ApprovalProgram))
+	copy(res.ApprovalProgram, ap.ApprovalProgram)
+	res.ClearStateProgram = make([]byte, len(ap.ClearStateProgram))
+	copy(res.ClearStateProgram, ap.ClearStateProgram)
 	res.GlobalState = ap.GlobalState.Clone()
 	return
 }
 
 // Clone returns a copy of some AppLocalState that may be modified without
 // affecting the original
-func (al AppLocalState) Clone() (res AppLocalState) {
-	res = al
+func (al *AppLocalState) Clone() (res AppLocalState) {
```

Comment:
Same here -- I would like to evaluate these pointer receiver changes in a separate PR so that we can measure any performance difference. If there isn't one, I don't want to do it.